src/main/java/ru/hedin/modelka/domain/ModUser.java

```diff
@@ -0,0 +1,94 @@
+package ru.hedin.modelka.domain;
+
+import org.springframework.security.core.GrantedAuthority;
+import ru.hedin.modelka.security.UserRole;
+
+import javax.persistence.*;
+import java.util.*;
+
+@Entity
+@Table(name = "mod_user")
+
+public class ModUser {
+    @Id
+  //  @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @SequenceGenerator(name="modelPKgen", sequenceName="MODEL_C_MODEL_seq")
+    @GeneratedValue(generator="modelPKgen", strategy= GenerationType.IDENTITY)
+    private long id;
+    @Column (nullable = false, unique = true )
+    private String userName;
+    @Column (nullable = false, unique = true )
+    private String password;
+    @Column (nullable = false )
+    private long longRoles = 1; //ROLE_USER
+    @Transient
+    private Set<UserRole> roles;
+//    @Bean
+//    public ModUser() {
+//    }
+
+
+    public long getLongRoles() {
+        return longRoles;
+    }
+
+    public void setLongRoles(long longRoles) {
+        this.longRoles = longRoles;
+    }
+    //calculation of longRoles
+    public void setLongRoles(Set<UserRole> roleSet) {
+        longRoles=0;
+        roles.forEach( (rol)->{
+            longRoles+= (long) Math.pow(2,rol.ordinal());
+        });
+
+    }
+
+    public  Set<UserRole>  getRoles() {
```

Comment:
https://stackoverflow.com/questions/2199399/storing-enumset-in-a-database

It struck me as a surprise that nobody was suggesting a well-maintained library instead of writing your own. The above answer is spot on and educational but it just encourages people to copy and paste code around (then mostly forget the credits).

Here's my 2 cents:

EnumSet<YourEnum> mySet = EnumSet.of(YourEnum.FIRST);
long vector = EnumUtils.generateBitVector(YourEnum.class, mySet);
EnumSet<YourEnum> sameSet = EnumUtils.processBitVector(YourEnum.class, vector);
See https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/EnumUtils.html

Reply:
Мы не храним в базе роли как роли, хранив в лонге как битовую на 64 роли запись

Reply:
Да и вот библиотека которая превращает роли в цыфру и обратно в 1 строку.